packages/ui/src/components/Button/Button.vue

```diff
@@ -1,56 +1,244 @@
 <script setup lang="ts">
-type BtnType = 'primary' | 'secondary'
+import { computed, ComputedRef } from 'vue'
+import { SSpinner } from '@/lib'
+
+const BUTTON_TYPE_VALUES = ['primary', 'secondary', 'outline', 'action'] as const
+const BUTTON_SIZE_VALUES = ['mini', 'small', 'medium', 'big'] as const
+const BUTTON_ICON_POSITION_VALUES = ['left', 'right'] as const
+
+type ButtonType = typeof BUTTON_TYPE_VALUES[number]
+type ButtonSize = typeof BUTTON_SIZE_VALUES[number]
+type ButtonIconPosition = typeof BUTTON_ICON_POSITION_VALUES[number]
+
+const SPINNER_SIZE: Record<ButtonSize, string> = {
+  mini: '12',
+  small: '18',
+  medium: '24',
+  big: '24',
+}
+
+const SPINNER_WIDTH: Record<ButtonSize, string> = {
+  mini: '3',
+  small: '4',
+  medium: '4',
+  big: '4',
+}
+
+function usePropTypeFilter<T>(value: T, validValues: readonly T[], defaultValue: T): ComputedRef<T> {
+  return computed(() => {
+    if (validValues.includes(value)) {
+      return value
+    }
+
+    return defaultValue
+  })
+}
 
 const props = withDefaults(
   defineProps<{
-    rounded?: boolean
-    type?: BtnType
+    type?: ButtonType,
+    size?: ButtonSize,
+    nativeType?: 'button' | 'reset' | 'submit',
+    icon?: string,
+    iconPosition?: ButtonIconPosition,
+    rounded?: boolean,
+    disabled?: boolean,
+    loading?: boolean,
+    uppercase?: boolean,
   }>(),
   {
-    type: 'primary',
+    type: 'secondary',
+    size: 'medium',
+    nativeType: 'button',
+    icon: '',
+    iconPosition: 'left',
+    rounded: false,
+    disabled: false,
+    loading: false,
+    uppercase: false,
   },
 )
+
+const definitelyType = usePropTypeFilter(props.type, BUTTON_TYPE_VALUES, 'primary')
+const definitelySize = usePropTypeFilter(props.size, BUTTON_SIZE_VALUES, 'medium')
+const definitelyIconPosition = usePropTypeFilter(props.iconPosition, BUTTON_ICON_POSITION_VALUES, 'left')
+const isAction = computed(() => definitelyType.value === 'action')
 </script>
 
 <template>
-  <div
-    role="button"
-    tabindex="0"
+  <button
+    :type="nativeType"
     :class="[
       's-button',
-      `s-button--type-${type}`,
+      `s-button_type_${definitelyType}`,
+      `s-button_size_${definitelySize}`,
+      `s-button_icon-position_${definitelyIconPosition}`,
```

Comment:
Please use standard [BEM naming](https://cssguidelin.es/#bem-like-naming) for modifiers, i.e. not `s-button_type_primary`, but `s-button--type-primary`.